spinalcordtoolbox/reports/qc2.py

```diff
@@ -273,6 +274,116 @@ def sct_register_multimodal(
         fig.savefig(img_path, format='png', transparent=True, dpi=300)
 
 
+def sct_deepseg(
+    fname_input: str,
+    fname_seg: str,
+    species: str,
+    argv: Sequence[str],
+    path_qc: str,
+    dataset: Optional[str],
+    subject: Optional[str],
+    fname_seg2: Optional[str],
+):
+    """
+    Generate a QC report for sct_deepseg, with varied colormaps depending on the type of segmentation.
+
+    This refactor is based off of the `listed_seg` method in qc.py, adapted to support multiple images.
+    """
+    command = 'sct_deepseg'
+    cmdline = [command]
+    cmdline.extend(argv)
+
+    # Axial orientation, switch between one anat image and 1-2 seg images
+    with create_qc_entry(
+        path_input=Path(fname_input).absolute(),
+        path_qc=Path(path_qc),
+        command=command,
+        cmdline=list2cmdline(cmdline),
+        plane='Axial',
+        dataset=dataset,
+        subject=subject,
+    ) as imgs_to_generate:
+        # FIXME: This code is more or less duplicated with the 'sct_register_multimodal' report, because both reports
+        #        use the old qc.py method "_make_QC_image_for_3d_volumes" for generating the background img.
```

Comment:
The amount of boilerplate and duplication between this `qc2.py` module and @mguaypaq's `sct_register_multimodal` addition is a little concerning to me. But, I'm hopeful that now we have a second data point, we can start to identify places where we can factor out the boilerplate.